troposphere/static/js/components/projects/detail/resources/ProjectDetails.react.js

```diff
@@ -82,53 +55,40 @@ define(function (require) {
       // Remove the resources from the list of selected resources
       selectedResources.remove(resource);
 
-      // If the resource is equal to the currently previewed resource, change
-      // the previewed resource to the last resource the user selected
-      if (previewedResource === resource) {
+      // Replace preview, with another
+      if (previewedResource == resource) {
         previewedResource = selectedResources.last();
       }
 
       this.setState({
-        selectedResource: previewedResource,
-        previewedResource: previewedResource,
-        selectedResources: selectedResources
+        previewedResource,
+        selectedResources,
       });
     },
 
-    onPreviewResource: function (resource) {
-      var selectedResources = this.state.selectedResources;
-
-      // todo: figure out why I did this...
-      selectedResources.reset();
-      selectedResources.push(resource);
-
-      this.setState({
-        selectedResource: resource,
-        previewedResource: resource,
-        selectedResources: selectedResources
-      });
+    onPreviewResource: function () {
+        this.deselectAllResources();
     },
 
     onMoveSelectedResources: function () {
-        var match = false,
-            attachedResources = stores.VolumeStore.getAttachedResources();
+        var attachedResources = stores.VolumeStore.getAttachedResources(),
+            anyAttached;
 
-        this.state.selectedResources.forEach(function(sel) {
-            if (attachedResources.indexOf(sel.get('uuid')) !== -1) {
-                match = true;
-            }
+        anyAttached = this.state.selectedResources.some(function(selected) {
+            return attachedResources.includes(selected.get('uuid'));
```

Comment:
I guess this confirms we can use [ES7 `includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) - am I correct? 

For reference, it appears that [`attachedResources` is just an Array](https://github.com/iPlantCollaborativeOpenSource/troposphere/blob/33f6fe01ef6f34ded880cd0a31f1b8b207d8e576/troposphere/static/js/stores/VolumeStore.js#L42-L53).

I definitely like having some variation of a _contain_ check available on Arrays.